### DISCUSSION TITLE
Open Document with ASN in Url

### DISCUSSION THREAD

**Discussion by Bart1909** [jonaswinkler/paperless-ng#807] [Support]
Hi,
I'm thinking of generating labels with a QR-Code on it. This QR-Code should contains the URL to Paperless with the ASN.
I could then stick the label on the documents before scanning them and after processing in Paperless (assigning the ASN to the document) I could scan the QR Code to get straight to the document.
At the moment, I only find a solution to open the document via url only with the internal document ID, not with the ASN.
I can only get the document number with the ASN via the API, but then, I have no redirect to the document GUI itself.

Is there a way to open the document directly with the ASN number or do I have to implement the redirection service on my own?

Thanks in advance!

**Comment by jonaswinkler** [comment]
There's currently no simple way to do that.

Maybe a redirect in the frontend that navigates from `<host>/asn/53` to `<host>/documents/<id>` could work.

**Reply by Bart1909** [reply]
I've had the same idea and tried to implement this on my own, but got stuck with the error described in #810 :D

**Reply by jonaswinkler** [reply]
I'll see what I can do about #810 this weekend.
